src/system_utils.jl

```diff
@@ -11,15 +11,16 @@ function _find_subnetworks(sys::PSY.System)
     branches = get_ac_branches(sys)
     @info "Validating connectivity with depth first search (network traversal)"
     M, bus_lookup = calculate_adjacency(branches, buses)
-    slack_positions = find_slack_positions(buses, bus_lookup)
-    return find_subnetworks(M, PSY.get_number.(buses)), slack_positions
+    ref_positions = find_slack_positions(buses, bus_lookup)
+    ref_buses = [k for (k, v) in bus_lookup if v in ref_positions]
+    return find_subnetworks(M, PSY.get_number.(buses)), ref_buses
 end
 
 """
 Finds the subnetworks in a system using Depth First Search (DFS). Returns a dictionary keyed
 by the reference bus of the subnetworks if they exist
 """
 function find_subnetworks(sys::PSY.System)
-    sbn, ref_bus_positions = _find_subnetworks(sys)
-    return assign_reference_buses(sbn, ref_bus_positions)
+    sbn, ref_buses = _find_subnetworks(sys)
+    return assign_reference_buses!(sbn, ref_buses)
```

Comment:
All the other existing `assign_reference_buses` calls have been edited to add a `bus_ax_ref` — this one doesn't need one?

Reply:
in some cases we already have the references buses so there is no need to recalculate them based on the ref_ax and location.